### PR TITLE
Add WYSIWYG & Block Editor Boilerplates

### PR DESCRIPTION
**Pull request by formfcw**
Should meet the requirements. I've added some examples of how to extend each of the editors in the README files.

### CODE REVIEW

**Review by ComfortablyCoding — CHANGES_REQUESTED** [review]
### Build Stage Issues
I receive the following warnings when attempting to build these extensions.

Screenshot 2024-08-26 at 4 34 25 PM
Screenshot 2024-08-26 at 4 36 26 PM

### Usage
Although the extensions are enabled and show under the extensions tab they do not show up in the `Interface` section for the respective fields

**Comment by formfcw** [comment]
> ### Build Stage Issues
> 
> I receive the following warnings when attempting to build these extensions.
> Screenshot 2024-08-26 at 4 34 25 PM Screenshot 2024-08-26 at 4 36 26 PM
> ### Usage
> 
> Although the extensions are enabled and show under the extensions tab they do not show up in the `Interface` section for the respective fields



I am afraid I can’t reproduce your issues. This is what it looks like when I build the extension:

terminal

And this is what the interface selector looks like:

interface-selector

Dockerfile:
```Dockerfile
COPY directus-labs/extensions/packages/input-rich-text-html/dist ./extensions/input-rich-text-html/dist
COPY directus-labs/extensions/packages/input-rich-text-html/package.json ./extensions/input-rich-text-html

COPY directus-labs/extensions/packages/input-block-editor/dist ./extensions/input-block-editor/dist
COPY directus-labs/extensions/packages/input-block-editor/package.json ./extensions/input-block-editor
```

What environment are you using?
Did you build the extensions before installing/adding them to the Directus container?

**Comment by ComfortablyCoding** [comment]
@formfcw 

They do not show up for me

Screenshot 2024-08-27 at 9 31 05 AM
Screenshot 2024-08-27 at 9 31 18 AM

> What environment are you using?
 
I am using a Mac M3, running directus via the `docker-compose.yml` [from the docs](https://docs.directus.io/self-hosted/docker-guide.html#example-docker-compose). The extensions are located in the `extensions` folder.

> Did you build the extensions before installing/adding them to the Directus container?

Yes, the build command is what shows those messages, they also show again when directus is started.

**Comment by formfcw** [comment]
So, I tried to reinstall everything to a new folder. So I tried to emulate your environment: While I am on a M1, I used the same `docker-compose.yml` [from the docs](https://docs.directus.io/self-hosted/docker-guide.html#example-docker-compose), as you did. Then I copied the extensions directly from the GitHub branch into the project folder, installed their dependencies and ran the `build` command.

Block Editor:
block-editor-build

WYSIWYG:
wysiwyg-build

(Yes, there is a (different) warning, that I don’t think we can fix, at the same time it builds successfully!)

Then I ran `docker compose up`, opened Directus, created a new collection and added the fields and interfaces via the selector:

interface-selector

And on the item page, they just work:

interfaces

So, I don’t think that I can reproduce your issues.
Can you (or someone else on your team) try to reinstall in the same way I described above?

**Comment by ComfortablyCoding** [comment]
> So, I tried to reinstall everything to a new folder. So I tried to emulate your environment: While I am on a M1, I used the same `docker-compose.yml` [from the docs](https://docs.directus.io/self-hosted/docker-guide.html#example-docker-compose), as you did. Then I copied the extensions directly from the GitHub branch into the project folder, installed their dependencies and ran the `build` command.
> 
> Block Editor:
> block-editor-build
> 
> WYSIWYG:
> wysiwyg-build
> 
> (Yes, there is a (different) warning, that I don’t think we can fix, at the same time it builds successfully!)
> 
> Then I ran `docker compose up`, opened Directus, created a new collection and added the fields and interfaces via the selector:
> 
> interface-selector
> 
> And on the item page, they just work:
> 
> interfaces
> 
> So, I don’t think that I can reproduce your issues.
> Can you (or someone else on your team) try to reinstall in the same way I described above?

These were the steps I did as well, I will see if someone else on the team can reproduce the same issues.

File: packages/input-rich-text-html/src/input-rich-text-html.vue [hunk 1]
```diff
@@ -0,0 +1,539 @@
...
+                    editorRef.value.editorCommands?.commands?.exec?.mcedirectionrtl();
+                } else {
+                    editorRef.value.editorCommands?.commands?.exec?.mcedirectionltr();
+                }
+            }
+        },
+    );
+
+    const editorOptions = computed(() => {
+        let styleFormats = null;
```
Comment (ComfortablyCoding) [inline packages/input-rich-text-html/src/input-rich-text-html.vue:161]:
```suggestion
        let styleFormats: CustomFormat[] | null = null;
```

File: packages/input-block-editor/README.md
Comment (ComfortablyCoding) [inline packages/input-block-editor/README.md:1]: I believe their is a missing step 5 which is to enable the Marker in the toolbar under the field interface options
 
Screenshot 2024-08-28 at 9 46 09 AM

**Review by ComfortablyCoding — CHANGES_REQUESTED** [review]
Re-trying with npm works and have been informed that is acceptable so we can disregard the build issues with pnpm.

File: packages/input-block-editor/README.md
Comment (formfcw) [reply packages/input-block-editor/README.md:1]: That is what second point in the README does:

https://github.com/directus-labs/extensions/blob/72e5f88d576e909f44c2cba14e24b258f4273649/packages/input-block-editor/README.md?plain=1#L10-L12

**Comment by formfcw** [comment]
> Re-trying (install and build) with npm works and have been informed that is acceptable so we can disregard the build issues with pnpm.

Oh, I didn’t know you were using `pnpm` … much ado about nothing

File: packages/input-block-editor/README.md
Comment (ComfortablyCoding) [reply packages/input-block-editor/README.md:1]: In that case I will retry this as it did not enable it for me after doing all the steps.

Comment (formfcw) [reply packages/input-block-editor/README.md:1]: > I believe we should add a step 5 which is to enable the Marker in the toolbar under the field interface options
> Screenshot 2024-08-28 at 9 46 09 AM

Look at the first list item :)

Comment (ComfortablyCoding) [reply packages/input-block-editor/README.md:1]: I did that manually

Comment (ComfortablyCoding) [reply packages/input-block-editor/README.md:1]: Based on investigation it is indeed not selected after the outlined steps. Adding `marker` to the `schema.default_values` for the `tools` field does resolve this issue.

Comment (formfcw) [reply packages/input-block-editor/README.md:1]: I have updated the README

**Review by ComfortablyCoding — DISMISSED** [review]
LGTM 🚀

**Review by phazonoverload — APPROVED** [review]

**Pull request merged**